cylc/flow/scripts/validate.py

```diff
@@ -89,8 +93,14 @@
         dest="profile_mode",
         sources={'validate'}
     ),
+    OptionSettings(
+        ["-u", "--run-mode"], help="Validate for run mode.", action="store",
+        default="live", dest="run_mode",
+        choices=['live', 'dummy', 'simulation']
+    ),
     VALIDATE_RUN_MODE,
-    VALIDATE_ICP_OPTION
+    VALIDATE_ICP_OPTION,
+    VALIDATE_AGAINST_SOURCE_OPTION,
```

Comment:
Not sure this is playing nicely with vip. I'm getting `ParsecError: Cannot validate .../cylc-src/workflow-name against source: it is not an installed workflow.` running vip with this `--against-source` option. Should we be removing it from the vip command?

Reply:
Probably worth a check - the against-source option is fundamentally antithetical to VIP. We should probably catch this error earlier.